docs/book/src/migration/migration_guide_v2tov3.md

```diff
@@ -0,0 +1,145 @@
+
+Next, we'll re-scaffold out the API types and controllers. Since we want both,
+we'll say yes to both the API and controller prompts when asked what parts we
+want to scaffold:
+
+```bash
+kubebuilder create api --group batch --version v1 --kind CronJob
+```
+
+<aside class="note warning">
+<h1>How to still keep `apiextensions.k8s.io/v1beta1` for CRDs?</h1>
+
+From now on, the CRDs that will be created by controller-gen will be using the Kubenetes API version `apiextensions.k8s.io/v1`  by default, instead of `apiextensions.k8s.io/v1beta1` . So, if you would like to keep using the previous version, use the flag `--crd-version=v1beta` in the above command.  
+
+</aside>
+
+<aside class="note warning">
+<h1>If you're using multiple groups</h1>
+
+Please run `kubebuilder edit --multigroup=true` to enable multi-group support before migrating the APIs and controllers. Please see [this](/migration/multi-group.md) for more details.
+
+</aside>
+
+### Migrate the APIs
+
+Now, let's copy the API definition from `api/v1/cronjob_types.go` from our old project to the new one. 
+
+### Migrate the Controllers
+
+Now, let's copy the controller code from `controllers/cronjob_controller.go` from our old project to the new one and fix the following breaking change. Replace:
+
+```go 
+func (r *CronJobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+ 	ctx := context.Background() 
+ 	log := r.Log.WithValues("cronjob", req.NamespacedName)
+```
+
+With:
+
+```go 
+func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("cronjob", req.NamespacedName)
+```
+
+<aside class="note warning">
+<h1>Controller-runtime version updated has breaking changes</h1>
+
+Check [sigs.k8s.io/controller-runtime release docs from 0.7.0+ version]( https://github.com/kubernetes-sigs/controller-runtime/releases ) for breaking changes.
+
+</aside>
+
+## Migrate the Webhooks
+
+<aside class="note warning">
+<h1>Note</h1>
+
+If you don't have any webhooks, you can skip this section.
+
+</aside>
+
+Now let's scaffold the webhooks for our CRD (CronJob). We'll need to run the
+following command with the `--defaulting` and `--programmatic-validation` flags
+(since our test project uses defaulting and validating webhooks):
+
+```bash
+kubebuilder create webhook --group batch --version v1 --kind CronJob --defaulting --programmatic-validation
+```
+
+<aside class="note warning">
+<h1>How to keep using `apiextensions.k8s.io/v1beta1` for Webhooks?</h1>
+
+From now on, the Webhooks that will be created by Kubebuilder using by default the Kubernetes API version `apiextensions.k8s.io/v1` instead of `apiextensions.k8s.io/v1beta1`,  `admissionregistration.k8s.io/v1` instead of `admissionregistration.k8s.io/v1beta1` and the `cert-manager.io/v1` to replace `cert-manager.io/v1alpha2`. So, if you would like to keep using the previous version use the flag `--webhook-version=v1beta` in the above command.  
```

Comment:
if you use the latest version of webhook than the new cert-manager will be used otherwise it should scaffold the legacy one. If it is not working in this way is a bug.